File: durmitor/src/main/java/ql/ast/expression/Binary.java
package ql.ast.expression;

public abstract class Binary extends Expression {
    
    protected Expression lhs;
    protected Expression rhs;
    
    public Binary(Expression lhs, Expression rhs) {
        this.lhs = lhs;
        this.rhs = rhs;
    }

    @Override
    public String toString()
    {
        return lhs + " op " + rhs;
    }
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QL/AST/Expression/Binary.cs
namespace Assignment1.Model.QL.AST.Expression
{
    public abstract class Binary : ASTNode, IExpression
    {
        public IExpression Left { get; }
        public IExpression Right { get; }

        protected Binary(IExpression left, IExpression right)
        {
            Left = left;
            Right = right;
        }

        public abstract void Accept(IExpressionVisitor visitor);
    }
}
